WireAnalytics/Sources/WireAnalytics/AnalyticsSession.swift

```diff
@@ -18,22 +18,82 @@
 
 import Countly
 
+/// A structure representing a user's profile for analytics purposes.
+
+public struct AnalyticsUserProfile {
+
+    /// The unique identifier for the user's analytics profile.
+    public let analyticsIdentifier: String
+
+    /// The identifier for the team the user belongs to.
+    public let teamID: String?
+
+    /// The role of the user within the team.
+    public let teamRole: String
+
+    /// The size of the team the user belongs to.
+    public let teamSize: Int?
+
+    /// The number of contacts the user has.
+    public let contactCount: Int?
+
+    public init(
+        analyticsIdentifier: String,
+        teamID: String? = nil,
+        teamRole: String,
+        teamSize: Int? = nil,
+        contactCount: Int? = nil
+    ) {
+
+        self.analyticsIdentifier = analyticsIdentifier
+        self.teamID = teamID
+        self.teamRole = teamRole
+        self.teamSize = teamSize
+        self.contactCount = contactCount
+
+    }
+
+}
+
 public struct AnalyticsSession: AnalyticsSessionProtocol {
 
     let countly: WireCountly
+    let appKey: String
+    let host: URL
+    private var userProfile: AnalyticsUserProfile?
 
-    public init(appKey: String, host: URL) {
+    public init(
+        appKey: String,
+        host: URL,
+        userProfile: AnalyticsUserProfile
+    ) {
+        self.appKey = appKey
+        self.host = host
```

Comment:
Is there a need to retain these?

Reply:
No. fixed it here: https://github.com/wireapp/wire-ios/commit/0d387898002335dbc9926f2994c1957025573898